cpp/modmesh/transform/fourier.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <modmesh/math.hpp>
+#include <modmesh/buffer/buffer.hpp>
+
+namespace modmesh
+{
+
+namespace transform
+{
+
+namespace detail
+{
+
+size_t bit_reverse(size_t n, const size_t bits);
+
+} /* end namespace detail */
+
+template <template <typename> typename T1, typename T2>
+void dft(SimpleArray<T1<T2>> const & in, SimpleArray<T1<T2>> & out)
```

Comment:
The template of template `<T1<T2>>` syntax looks so remote to me, although I understand why it is used.  Is this the only way to support complex number?

Reply:
hmm...., I need more time to think about it... 
This way was the first thing that came to my mind, although it looks quite complicated.

Reply:
OK, we can live with it in this PR and improve later.